code-challenges.test.js

```diff
@@ -37,10 +63,33 @@ var randomNouns1 = ["streetlamp", "potato", "teeth", "conclusion", "nephew"]
 var randomNouns2 = ["temperature", "database", "chopsticks", "mango", "deduction"]
 // Expected output: ["Temperature", "Database", "Chopsticks", "Mango", "Deduction"]
 
+// 2a) Pseudo code
+// Create describe, it, and expect statements to describe the problem statement
+// Add appropriate inputs/outputs, with one expect line per input/output pair
+// Run the test on terminal (yarn jest) prior to creating the function to ensure it turns red (note: if set up correctly, should get an error saying the function is not defined)
 
+describe("capitalizer", () => {
+    it("a function that takes in an array with strings and return another array with all strings capitalized", () => {
+      expect(capitalizer(randomNouns1)).toEqual(["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"])
+      expect(capitalizer(randomNouns2)).toEqual(["Temperature", "Database", "Chopsticks", "Mango", "Deduction"])
+    })
+  })
 
 // b) Create the function that makes the test pass.
 
+// 2b) Pseudo code:
+// Declare a function and name it capitalizer (const declaration)
+// Name the parameter as array
+// Create a return statement which uses HOF .map() to create a new array
+// .map() logic
+  // value.slice(0, 1).toUpperCase() + value.slice(1, value.length) (this will concatenate the slice of the first letter and then add in the rest of the string via slice)
+
+const capitalizer = (array) => {
+    return array.map(value => 
+        value.slice(0, 1).toUpperCase() + value.slice(1, value.length))
+    }
+
+// 2) Notes: Ran the code w/o a function and it was red. I added the function in and ran the test, got it passing after a few tries. For refactoring, I think .map() is probably the most efficient method to use. I could use either a for loop or .forEach(), but both of these methods seem like they would take considerably more code. I could also potentially split() the array, concatenate and slice as seen above, and then .join() the result. But this also seems like it'd be less efficient than .map(), which was specifically created for this purpose.
```

Comment:
Map is definitely the right choice here. A refactor could be removing the `value.length` from the second slice as it will default to the length.